integration-tests/src/test_loop/utils/resharding.rs

```diff
@@ -0,0 +1,487 @@
+use std::cell::Cell;
+use std::rc::Rc;
+
+use assert_matches::assert_matches;
+use itertools::Itertools;
+use near_async::test_loop::data::TestLoopData;
+use near_client::{Query, QueryError::GarbageCollectedBlock};
+use near_crypto::Signer;
+use near_primitives::test_utils::create_user_test_signer;
+use near_primitives::transaction::SignedTransaction;
+use near_primitives::types::{AccountId, BlockId, BlockReference, Gas};
+use near_primitives::views::{
+    FinalExecutionStatus, QueryRequest, QueryResponse, QueryResponseKind,
+};
+use rand::seq::SliceRandom;
+use rand::{Rng, SeedableRng};
+use rand_chacha::ChaCha20Rng;
+
+use super::sharding::this_block_has_new_shard_layout;
+use crate::test_loop::env::TestData;
+use crate::test_loop::utils::loop_action::LoopAction;
+use crate::test_loop::utils::sharding::next_block_has_new_shard_layout;
+use crate::test_loop::utils::transactions::{
+    check_txs, delete_account, get_anchor_hash, get_next_nonce, store_and_submit_tx, submit_tx,
+};
+use crate::test_loop::utils::{get_node_data, retrieve_client_actor, ONE_NEAR, TGAS};
+
+// Returns a callable function that, when invoked inside a test loop iteration, can force the creation of a chain fork.
+#[cfg(feature = "test_features")]
+pub(crate) fn fork_before_resharding_block(double_signing: bool) -> LoopAction {
+    use near_client::client_actor::AdvProduceBlockHeightSelection;
+
+    let done = Rc::new(Cell::new(false));
```

Comment:
Not sure if intentional, but you might have lost the changes to use `LoopAction::shared_success_flag();`

Reply:
Good catch! Yeah, missed that :)

Reply:
fixed in 19a382c585b68eb81a40704fc93913c42b522dd2